auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java

```diff
@@ -23,30 +23,45 @@
 public class AsyncSecurityCheckExecutor extends AbstractSecurityCheckExecutor<AsyncSecurityCheckExecutor> {
 
     private final static String TAG = "AsyncSecurityCheckExecutor";
+    private final static Logger LOG = MobileCore.getLogger();
 
     private final ExecutorService executorService;
-    private final static Logger LOG = MobileCore.getLogger();
 
 
+    /**
+     * Builder class for constructing an AsyncSecurityCheckExecutor object.
+     */
     public static class Builder extends SecurityCheckExecutor.Builder.AbstractBuilder<Builder, AsyncSecurityCheckExecutor> {
 
         private ExecutorService executorService;
 
+        /**
+         * Creates a Builder object.
+         *
+         * @param ctx {@link Context} to be used by security checks
+         * @throws IllegalArgumentException if ctx is null
+         */
         Builder(final Context ctx) {
             super(ctx);
         }
 
         /**
          * Specify a custom execution singleThreadService for this SecurityCheckExecutor.
          *
-         * @param executorService executor singleThreadService to be used.
+         * @param executorService {@link AppExecutors#singleThreadService()} to be used. Can be null
          * @return this
          */
         public Builder withExecutorService(@Nullable final ExecutorService executorService) {
             this.executorService = executorService;
             return this;
         }
 
+        /**
+         * Creates a new AsyncSecurityCheckExecutor object.
+         * If no {@link ExecutorService} has been defined, a new {@link AppExecutors#singleThreadService()} is created.
```

Comment:
`If no {@link ExecutorService} has been defined, defaults to {@link AppExecutors#singleThreadService()}.`